driver-core/src/main/java/com/datastax/driver/core/NettyOptions.java

```diff
@@ -0,0 +1,183 @@
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+/**
+ * A set of options, or rather hooks, that allow clients to customize the driver's underlying Netty layer.
+ * <p>
+ * Clients that need to hook into the driver's underlying Netty layer can
+ * subclass this class and provide the necessary customization by overriding
+ * its methods.
+ * <p>
+ * Typically, clients would register this class with {@link Cluster#builder()}:
+ *
+ * <pre>
+ *     NettyOptions nettyOptions = ...
+ *     Cluster cluster = Cluster.builder()
+ *          .addContactPoint(...)
+ *          .withNettyOptions(nettyOptions)
+ *          .build();
+ * </pre>
+ *
+ * <strong>Extending the NettyOptions API</strong>
+ * <p>
+ * Contrary to other driver options, the options available in this class should
+ * be considered as advanced features and as such, <em>they should only be
+ * modified by expert users</em>.
+ * <p>
+ * <strong>A misconfiguration introduced by the means of this API can have unexpected results
+ * and cause the driver to completely fail to connect.</strong>
+ * <p>
+ * Moreover, since versions 2.0.9 and 2.1.4 (see JAVA-538),
+ * the driver is available in two different flavors: with a standard Maven dependency on Netty,
+ * or with a "shaded" (internalized) Netty dependency.
+ * <p>
+ * Given that NettyOptions API exposes Netty classes ({@link SocketChannel}, etc.),
+ * <em>it should only be extended by clients using the non-shaded
+ * version of driver</em>.
+ * <p>
+ * <strong>Extending this API with shaded Netty classes is not supported,
+ * and in particular for OSGi applications, it is likely that such a configuration would lead to
+ * compile and/or runtime errors.</strong>
+ *
+ * @jira_ticket JAVA-640
+ * @since 2.0.10
+ */
+public class NettyOptions {
+
+    /**
+     * The default instance of {@link NettyOptions} to use.
+     */
+    public static final NettyOptions DEFAULT_INSTANCE = new NettyOptions();
+
+    /**
+     * Create and return a new {@link EventLoopGroup} instance to use.
+     * <p>
+     * This hook is invoked only once at Cluster initialization;
+     * the returned instance will be kept in use throughout the cluster lifecycle.
+     * <p>
+     * The default implementation will create an instance of {@link EpollEventLoopGroup}
+     * if {@link NettyUtil#isEpollAvailable() epoll is available},
+     * or {@link NioEventLoopGroup} otherwise.
+     *
+     * @param threadFactory The {@link ThreadFactory} to use with the new {@link EventLoopGroup} instance;
+     *                      The driver will provide its own internal thread factory here.
+     *                      It is safe to ignore it and use another thread factory.
+     * @return the newly-created {@link EventLoopGroup} instance to use.
+     */
+    public EventLoopGroup createEventLoopGroup(ThreadFactory threadFactory) {
+        return NettyUtil.isEpollAvailable() ? new EpollEventLoopGroup(0, threadFactory): new NioEventLoopGroup(0, threadFactory);
```

Comment:
:+1: looks nice and clean, also like how you can just call the constructor directly here instead of using newInstance() like it was before.  Didn't occur to me at the time that this would work but makes sense.